sidecar/main.go

```diff
@@ -106,11 +107,22 @@ func main() {
 		PodNamespace: *podNamespace,
 		PodUID:       *podUID,
 	}
-	err = nodeMgr.Deploy()
+
+	nodeObj, err := nodeMgr.GetCSIAddonsNode()
 	if err != nil {
-		klog.Fatalf("Failed to create csiaddonsnode: %v", err)
+		klog.Fatalf("failed to get csiaddonsNode object: %v", err)
 	}
 
+	// Start the watcher, it calls nodeMgr.Deploy() internally
+	go func() {
+		err := nodeMgr.DispatchWatcher(nodeObj)
+		if err != nil {
+			klog.Errorf("Watcher for %s failed due to error: %v", nodeObj.Name, err)
+
+			os.Exit(1)
```

Comment:
Why not use [`klog.Fatalf()`](https://pkg.go.dev/k8s.io/klog#Fatalf) instead of adding the `os.Exit()`?